src/codecs/png.rs

```diff
@@ -131,6 +132,8 @@ impl<R: Read> PngDecoder<R> {
         let info = decoder.read_header_info().map_err(ImageError::from_png)?;
         limits.check_dimensions(info.width, info.height)?;
 
+        let icc_profile = info.icc_profile.as_ref().map(|x| x.to_vec());
```

Comment:
This will always be None. `read_header_info` never returns ICC information, you have to call `info()` on the reader returned from `read_info()` instead.

Reply:
Ah, thanks. I don't even have to store it. Fixed.